user_notification/models.py

```diff
@@ -8,16 +10,25 @@ class JobAlert(models.Model):
                                         on_delete=models.CASCADE,
                                         blank=True,
                                         null=True)
-    alert_message = models.TextField()
-    alert_frequency = models.CharField(max_length=20)
+    last_check_date = models.DateField(default=datetime.today)
+    frequency_in_days = models.IntegerField(default=1)
     job_alert_type = models.CharField(max_length=50)
-    job_alert_scope = models.CharField(max_length=50)
-    job_alert_city = models.CharField(max_length=50)
-    job_alert_company_name = models.CharField(max_length=20)
+    job_alert_scope = models.CharField(max_length=50, choices=JobScope.choices, default=JobScope.UNSPECIFIED,
+                                       blank=True, null=True)
+    job_alert_city = models.CharField(max_length=50, blank=True, null=True)
+    job_alert_company_name = models.CharField(
+        max_length=20, blank=True, null=True)
 
     def __str__(self):
         return str(self.user_account_id)
 
+    # Calculate the date of the next alert
+    @classmethod
+    def calc_check_date(cls, last_check_date, frequency_in_days):
+        temp = datetime.strptime(last_check_date, "%Y-%m-%d").date()
```

Comment:
what will happen if frequency_in_days would be a negative number?

Reply:
I'll change the model field so it will set as PositiveIntegerField

Reply:
:+1:

Reply:
Also change the parameter name "temp" to be more meaningful

Reply:
fixed